substrate/utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -509,21 +483,241 @@ impl PalletCmd {
+		let executor = WasmExecutor::<(
+			sp_io::SubstrateHostFunctions,
+			frame_benchmarking::benchmarking::HostFunctions,
+			F,
+		)>::builder()
+		.with_execution_method(method)
+		.with_allow_missing_host_functions(self.allow_missing_host_functions)
+		.build();
+
+		let runtime = self.runtime_blob(&state)?;
+		let runtime_code = runtime.code()?;
+
+		let genesis_json: Vec<u8> = Self::exec_state_machine(
+			StateMachine::new(
+				&state,
+				&mut Default::default(),
+				&executor,
+				"GenesisBuilder_create_default_config",
+				&[], // no args for this call
+				&mut Self::build_extensions(executor.clone()),
+				&runtime_code,
+				CallContext::Offchain,
+			),
+			"build the genesis spec",
+		)?;
+
+		// Sanity check that it is JSON before we plug it into the next call.
+		if !serde_json::from_slice::<serde_json::Value>(&genesis_json).is_ok() {
+			log::warn!("GenesisBuilder::create_default_config returned invalid JSON");
+		}
+
+		let mut changes = Default::default();
+		let build_config_ret: Vec<u8> = Self::exec_state_machine(
+			StateMachine::new(
+				&state,
+				&mut changes,
+				&executor,
+				"GenesisBuilder_build_config",
+				&genesis_json.encode(),
+				&mut Extensions::default(),
+				&runtime_code,
+				CallContext::Offchain,
+			),
+			"populate the genesis state",
+		)?;
+
+		if !build_config_ret.is_empty() {
+			log::warn!("GenesisBuilder::build_config returned unexpected data");
+		}
+
+		Ok(changes)
+	}
+
+	/// Execute a state machine and decode its return value as `R`.
+	fn exec_state_machine<R: Decode, H: Hash, Exec: CodeExecutor>(
+		mut machine: StateMachine<BenchmarkingState<H>, H, Exec>,
+		hint: &str,
+	) -> Result<R> {
+		let res = machine
+			.execute()
+			.map_err(|e| format!("Could not call runtime API to {hint}: {}", e))?;
+		let res = R::decode(&mut &res[..])
+			.map_err(|e| format!("Failed to decode runtime API result: {:?}", e))?;
+		Ok(res)
+	}
+
+	/// Build the extension that are available for pallet benchmarks.
+	fn build_extensions<E: CodeExecutor>(exe: E) -> Extensions {
+		let mut extensions = Extensions::default();
+		let (offchain, _) = TestOffchainExt::new();
+		let (pool, _) = TestTransactionPoolExt::new();
+		let keystore = MemoryKeystore::new();
+		extensions.register(KeystoreExt::new(keystore));
+		extensions.register(OffchainWorkerExt::new(offchain.clone()));
+		extensions.register(OffchainDbExt::new(offchain));
+		extensions.register(TransactionPoolExt::new(pool));
+		extensions.register(ReadRuntimeVersionExt::new(exe));
+		extensions
+	}
+
+	/// Load the runtime blob for this benchmark.
+	///
+	/// The blob will either be loaded from the `:code` key out of the chain spec, or from a file
+	/// when specified with `--runtime`.
+	fn runtime_blob<'a, H: Hash>(
+		&self,
+		state: &'a BenchmarkingState<H>,
+	) -> Result<FetchedCode<'a, BenchmarkingState<H>, H>> {
+		if let Some(runtime) = &self.runtime {
+			log::info!("Loading WASM from {}", runtime.display());
```

Comment:
is it intentional that you don't pass log target? I generally find them easier to filter. Default is it will use the crate name or something?

Reply:
The log target is just the crate name. Not sure it adds much to make it more fine-grained.